Modules/System Tests/Cryptography Management/src/X509CertificateTest.Codeunit.al

```diff
@@ -0,0 +1,258 @@
+
+        // [WHEN] Verify Certificate from Base64 value 
+        CertificateVerified := X509CertificateCryptography.VerifyCertificate(CertBase64Value, '', X509ContentType::Pkcs12);
+
+        // [THEN] Verify that certificate is created
+        LibraryAssert.IsTrue(CertificateVerified, 'Failed to verify certificate.');
+    end;
+
+    [Test]
+    procedure VerifyCertificateFriendlyNameFromBase64Cert()
+    var
+        X509ContentType: Enum "X509 Content Type";
+        CertBase64Value: Text;
+        FriendlyName: Text;
+    begin
+        // [SCENARIO] Create certificate from Base64, and verify FriendlyName from certificate
+        // [GIVEN] Get Test Certificate Base64 value
+        CertBase64Value := GetCertificateBase64();
+
+        // [WHEN]  Get Certificate FriendlyName
+        X509CertificateCryptography.GetCertificateFriendlyName(CertBase64Value, '', FriendlyName);
+
+        // [THEN] Verify Results
+        LibraryAssert.AreEqual(FriendlyName, GetFriendlyName(), 'Failed to create certificate.');
+    end;
+
+    [Test]
+    procedure VerifyCertificateThumbprintFromBase64Cert()
+    var
+        X509ContentType: Enum "X509 Content Type";
+        CertBase64Value: Text;
+        Thumbprint: Text;
+    begin
+        // [SCENARIO] Create certificate from Base64, and verify Thumbprint from certificate
+        // [GIVEN] Get Test Certificate Base64 value
+        CertBase64Value := GetCertificateBase64();
+
+        // [WHEN]  Get Certificate Thumbprint
+        X509CertificateCryptography.GetCertificateThumbprint(CertBase64Value, '', Thumbprint);
+
+        // [THEN] Verify Results        
+        LibraryAssert.AreEqual(Thumbprint, GetThumbprint(), 'Failed to create certificate.');
+    end;
+
+    procedure VerifyCertificateIssuerFromBase64Cert()
+    var
+        X509ContentType: Enum "X509 Content Type";
+        CertBase64Value: Text;
+        Issuer: Text;
+    begin
+        // [SCENARIO] Create certificate from Base64, and verify Issuer from certificate
+        // [GIVEN] Get Test Certificate Base64 value
+        CertBase64Value := GetCertificateBase64();
+
+        // [WHEN]  Get Certificate Issuer
+        X509CertificateCryptography.GetCertificateIssuer(CertBase64Value, '', Issuer);
+
+        // [THEN] Verify Results        
+        LibraryAssert.AreEqual(Issuer, GetIssuer(), 'Failed to create certificate.');
+    end;
+
+    [Test]
+    procedure VerifyCertificateExpirationFromBase64Cert()
+    var
+        X509ContentType: Enum "X509 Content Type";
+        CertBase64Value: Text;
+        Expiration: DateTime;
+    begin
+        // [SCENARIO] Create certificate from Base64, and verify Expiration Date from certificate
+        // [GIVEN] Get Test Certificate Base64 value
+        CertBase64Value := GetCertificateBase64();
+
+        // [WHEN]  Get Certificate Expiration
+        X509CertificateCryptography.GetCertificateExpiration(CertBase64Value, '', Expiration);
+
+        // [THEN] Verify Results        
```

Comment:
Let's make [THEN] step it in the form "Certificate Expiration is retrieved" for this one and all similar tests